sim/midas/src/main/cc/core/simulation.cc

```diff
@@ -125,13 +125,15 @@ void simulation_t::wait_for_init() {
 }
 
 void simulation_t::init_dram() {
-  auto &loadmem = registry.get_widget<loadmem_t>();
-  if (do_zero_out_dram) {
-    fprintf(stderr, "Zeroing out FPGA DRAM. This will take a few seconds...\n");
-    loadmem.zero_out_dram();
-  }
+  if (auto *loadmem = registry.get_widget_opt<loadmem_t>()) {
+    if (do_zero_out_dram) {
+      fprintf(stderr,
+              "Zeroing out FPGA DRAM. This will take a few seconds...\n");
```

Comment:
It might be wise to print something to the effect of: "Skipping DRAM zero-initialization since FPGA DRAM is unused" or something to that effect in an else.